lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -99,60 +99,62 @@ define((require, exports, module) ->
       rangeCache: {}
 
       compareTokens: (token1, token2) ->
-        token1["type"] == token2["type"] and token1["value"] == token2["value"]
+        (not token1? and not token2?) or token1? and token2? and token1.type == token2.type and token1.value == token2.value
 
       getEquationStart: (tokenIterator) ->
-        j = erh.BEGIN_EQUATION_TOKEN_SEQUENCE.length - 1
-        curEquationStart = null
-        # in case we're actually inside `\begin{equation}`
+        # following cycle pushes tokenIterator to the end of
+        # beginning sequence, if it is inside one
         for token in erh.BEGIN_EQUATION_TOKEN_SEQUENCE
           if erh.compareTokens(token, tokenIterator.getCurrentToken())
             tokenIterator.stepForward()
-        while j >= 0
-          if erh.compareTokens(erh.BEGIN_EQUATION_TOKEN_SEQUENCE[j], tokenIterator.stepBackward())
-            if j == erh.BEGIN_EQUATION_TOKEN_SEQUENCE.length - 1
+        curSequenceIndex = erh.BEGIN_EQUATION_TOKEN_SEQUENCE.length - 1
+        curEquationStart = null
+        while curSequenceIndex >= 0
+          if erh.compareTokens(erh.BEGIN_EQUATION_TOKEN_SEQUENCE[curSequenceIndex], tokenIterator.stepBackward())
```

Comment:
pls insert a linebreak after the opening bracket `compareTokens` (and maybe after the first argument too)